### ISSUE TITLE
false warning implicitly declaring library function `strlcpy`

### ISSUE THREAD

**Issue by llvmbot** [llvm/llvm-project#21194]
|  |  |
| --- | --- |
| Bugzilla Link | [20820](https://llvm.org/bz20820) |
| Version | 3.4 |
| OS | Linux |
| Reporter | LLVM Bugzilla Contributor |
| CC | @majnemer,@DougGregor |

## Extended Description 
```
clang -O2 -Wall -Wmissing-prototypes -Wpointer-arith -Wdeclaration-after-statement -Wendif-labels -Wmissing-format-attribute -Wformat-security -fno-strict-aliasing -fwrapv -g -DFRONTEND -I../../src/include -D_GNU_SOURCE -I/usr/include/libxml2   -c -o exec.o exec.c
exec.c:280:2: warning: implicitly declaring library function 'strlcpy' with type 'unsigned long
      (char *, const char *, unsigned long)'
        strlcpy(link_buf, fname, sizeof(link_buf));
        ^
exec.c:280:2: note: please include the header <string.h> or explicitly provide a declaration for 'strlcpy'
1 warning generated.
```
It can be tested on Postgres master sources.

**Comment by AaronBallman** [comment]
This is no longer an issue; autoconf was fixed but also, Clang now diagnoses this as an error by default (though it can be downgraded into a warning). Closing this as completed.
